### DISCUSSION TITLE
articles/the-history-of-software-engineering

### DISCUSSION THREAD

**Discussion by giscus** [hutusi/hutusi.github.com#191] [Comments]
# articles/the-history-of-software-engineering

TL;DR 本文13200+字，全文阅读约需20分钟。其中2680字的示例部分为ChatGPT生成，因此可以说本文GPT含量20%，不过该部分详细内容与本文主旨关系不大，且由GPT-4生成，可略过不看。本文简要回顾了从软件及计算机诞生到当前大模型AI时代的软件工程发展历史，并试图从软件工程历史和ChatGPT实践案例中探讨在大模型时代的软件开发模式。

https://hutusi.com/articles/the-history-of-software-engineering

**Comment by yexia553** [comment]
又是一篇深度好文，感叹作者的阅读、信息分析和写作的能力！

**Comment by weekend-project-space** [comment]
以前写程序是人类通过事物的观察及对信息理解把规则提取出来编写成代码的线条和脉络，也就是说是重复且机械化的操作。
AI 这种则是省略了观察和理解 直接给其信息，省略掉了中间的提取和规则编写，我们对其得出结果中间的组合可预测性更低。
关于模型的能力的边界，现在我对AI模型的理解还停留在利用已有的数据进行统计更好的预测下一个词，因此它在未知和数据不足的情况下没有办法完全超越过往已知的数据，但是从长远看，如果它成长之后拥有更强的推理能力，则跨过了已知的边界，而那些将会是科学和探索的前沿，也许在这个时代我们会看到不论从知识总量还是思考推理深度都全面超越人这个物种的智能体的出现，希望它能赡养人类这个物种，作为个体自由意志是否还存在也许就不会在被探讨了。也许会有新的议题，也许是美丽新世界，未来发展方向谁又能完全预测呢

**Reply by Catheryan** [reply]
并不赞同，未来Ai并不是银弹

**Comment by scriptway** [comment]
哈哈 我已经是一个熟练使用ai的新时代程序员了

**Comment by laipz8200** [comment]
如果LLM的知识完全来自人类已经存在的知识的排列组合，那我实在无法认同它能够对软件开发的本质性工作有什么改变……或许它能够帮助不那么熟练的工程师用已经存在的方法做抽象从而在一定程度上提高效率，但这好像并不是什么大不了的变化？

**Comment by simplecode101** [comment]
看完了，忍不住上来鼓励一下作者，值得收藏的好文章

**Reply by hutusi** [reply]
谢谢，谢谢大家：）
